workflow-templates/publish-go-tester-task.yml

```diff
@@ -39,10 +39,12 @@ jobs:
         id: determination
         run: |
           RELEASE_BRANCH_REGEX="refs/heads/[0-9]+.[0-9]+.x"
+          TAG_REGEX="refs/tags/.*"
           # The `create` event trigger doesn't support `branches` filters, so it's necessary to use Bash instead.
           if [[
             "${{ github.event_name }}" != "create" ||
-            "${{ github.ref }}" =~ $RELEASE_BRANCH_REGEX
+            "${{ github.ref }}" =~ $RELEASE_BRANCH_REGEX && \
+            ! "${{ github.ref }}" =~ $TAG_REGEX \
```

Comment:
[`&&` has higher precedence than `||`](https://tldp.org/LDP/abs/html/opprecedence.html) so the parentheses on the first expression (as you configured [the workflow in the `arduino/arduino-cli` repo](https://github.com/arduino/arduino-cli/blob/53a276832f017f1424c563d3c5d9b74d1000e93c/.github/workflows/publish-go-tester-task.yml#L45-L46)) is required.

A minimal demonstration of the logic without the parentheses:

```text
if [[ "a" == "a" || "a" == "b" && "a" == "b" ]]; then echo "Test evaluated true"; else echo "Test evaluated false"; fi
Test evaluated true
```

with parentheses:

```text
$ if [[ ("a" == "a" || "a" == "b") && "a" == "b" ]]; then echo "Test evaluated true"; else echo "Test evaluated false"; fi
Test evaluated false
```